src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -886,19 +880,21 @@ public override void DidReceiveChallenge (NSUrlSession session, NSUrlSessionTask
 #endif
 				var trustCallbackForUrl = sessionHandler.TrustOverrideForUrl;
 #if NET
-				var hasCallBack = trustCallbackForUrl is not null;
+				var hasCallBack = trustCallbackForUrl is not null || sessionHandler.ServerCertificateCustomValidationCallback is not null;
 #else
-				var hasCallBack = trustCallback is not null || trustCallbackForUrl is not null;
+				var hasCallBack = trustCallback is not null || trustCallbackForUrl is not null || sessionHandler.ServerCertificateCustomValidationCallback is not null;
 #endif
 				if (hasCallBack && challenge.ProtectionSpace.AuthenticationMethod == NSUrlProtectionSpace.AuthenticationMethodServerTrust) {
 #if NET
 					// if the trust delegate allows to ignore the cert, do it. Since we are using nullables, if the delegate is not present, by default is false
-					var trustSec = (trustCallbackForUrl?.Invoke (sessionHandler, inflight.RequestUrl, challenge.ProtectionSpace.ServerSecTrust) ?? false);
+					var trustSec = (trustCallbackForUrl?.Invoke (sessionHandler, inflight.RequestUrl, challenge.ProtectionSpace.ServerSecTrust) ?? false) ||
+						(InvokeServerCertificateCustomValidationCallback (inflight.Request, challenge.ProtectionSpace.ServerSecTrust));
```

Comment:
What happens if both are set? Do we trust the native one of the manage APIs? Should we always trust first the results from ServerCertificateCustomValidationCallback ?

Reply:
I opted to trust the certificate if either of the callbacks trusts the certificate. It's basically the same in the #else branch. I guess the comment should be updated as well. Unless the callbacks contain some weird side effects, the order in which they are invoked doesn't matter. I'm not sure there's really a much better way of doing it.